lib/styles/trackInputModality.js

```diff
@@ -115,7 +115,12 @@ function setUpEventHandlers(disableFocusRingByDefault) {
         (hadKeyboardEvent && focusTriggersKeyboardModality(e.target)) ||
         (focusSetExplicitly(e.target) && !hadClickEvent)
       ) {
+        // both the JS state and the body attribute for keyboard modality
         globalThemeState.inputModality = 'keyboard';
+        document.body.setAttribute('modality', 'keyboard');
+      } else {
+        globalThemeState.inputModality = null;
+        document.body.setAttribute('modality', '');
```

Comment:
See https://github.com/learningequality/kolibri/issues/4822#issuecomment-458761889 and https://github.com/learningequality/kolibri/pull/4847 for some historical context on how the code got into its current state.